### DISCUSSION TITLE
Backup of VW database (Docker.raw)

### DISCUSSION THREAD

**Discussion by bigpal4** [dani-garcia/vaultwarden#1832] [Q&A]
I see tutorials for the backup of the database using sqlite3, but I don't see anything on how to backup the Docker.raw database, which I have set at the minimum of 8GB... I don't see any other place where docker is storing data.. it seems the locations I provided during the setup command were seemingly ignored.

Am I looking in the wrong place? (/Users/me/Library/Containers/com.docker.docker/Data/vms/0/data/)

How can I backup this 8GB file without copying the whole thing around? Exporting the database isn't useful for me, as I am hosting multiple users and I want to backup their data also.

**Comment by RealOrangeOne** [comment]
That's not vaultwarden, that's docker itself. on macOS, docker runs in a VM. What your looking at is the virtual disk file for that entire VM. If you want to back that up, it's just a file so copy it, but depending on how you have vaultwarden setup, that may or may not backup vaultwarden itself. Either way, backing that up is probably out of scope for here.

**Comment by bigpal4** [comment]
I found a suitable solution to the problem by backing up the docker container using the following two docker commands:

docker commit -p _instance_Id_ _instance_Name_
docker save -o _something_.tar _instance_Name_

I can then gzip that tarball on the local filesystem and then copy anywhere I want, including an iCloud folder. Better yet, I made a crontab that launches a simple shell script that runs these commands to capture a nightly backup.

If anyone has a better solution, by all means.

**Reply by jjlin** [reply]
See https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault.

Taking a snapshot of a Docker container can work, but the backup is way larger than it needs to be.
